SRA_Sensors/SRA_Sensors.ino

```diff
@@ -1,60 +1,89 @@
-#include <stdio.h> 
-#include <stdlib.h>
-#include "RoveComm.h"
 #include "SRA_Sensors.h"
 
-int timing; //Variable the keeps count to make sure sensor data is only sent every second
+int timing;   //variable to keep count
+
+RoveCommEthernet RoveComm; //what is this?
+rovecomm_packet packet;
+
+EthernetServer TCPServer(RC_ROVECOMM_SCIENCESENSORSBOARD_PORT);
+
+void setup()
+{
+  timing = 0;   //Initialize the timing variable
+
+  Computer_Serial.begin(115200);
+  CO2_Serial.begin(19200);
+  O2_Serial.begin(9600);  
+  CH4_Serial.begin(115200);
+  myusb.begin();
+  userial.begin(115200);
+
+  //start RoveComm
+  RoveComm.begin(RC_SCIENCESENSORSBOARD_FOURTHOCTET, &TCPServer, RC_ROVECOMM_SCIENCESENSORSBOARD_MAC);
+
+  pinMode(UVLED_ENABLE_PIN,OUTPUT); //setup UVLED
+  pinMode(Laser1,OUTPUT);
+  pinMode(Laser2,OUTPUT);
+  pinMode(Laser3,OUTPUT);
+
+  digitalWrite(Laser1, LOW);
+  digitalWrite(Laser2, LOW);
+  digitalWrite(Laser3, LOW);
+
+  // configure PIN mode
+  pinMode(NO2_CS, OUTPUT);
+
+  // set initial PIN state
+  digitalWrite(NO2_CS, HIGH);
+
+  // initialize SPI interface for MCP3208
+  //SPISettings settings(ADC_CLK, MSBFIRST, SPI_MODE0);
+  //SPI.begin();
+  //SPI.beginTransaction(settings);
 
-RoveCommEthernetUdp RoveComm;
-void setup(){
-  timing = 0;//Initialize the timing variable
-  
-  //start serial connection
-  Computer_Serial.begin(9600);//Computer
-  CO2_Serial.begin(19200);//CO2
-  //RoveCom_Serial.begin(115200);//RoveCom
-  O2_Serial.begin(9600);//O2
-  Methane_Serial.begin(115200); //Methane
-  
-  //Start RoveComm
-  RoveComm.begin(RC_SCIENCESENSORSBOARD_FOURTHOCTET);
-  
-  pinMode(UVLED_ENABLE_PIN,OUTPUT); //Setup UvLed
-  
   delay(100);
 }
 
-void loop(){
-  //Read Sensor data only every second
+void loop()
+{
+  //Read sensor data only every second
+  myusb.Task();
   if(timing%10==0)
   {
-    co2Reading();
     o2Reading();
+    co2Reading();
+    //noReading();
+    //no2Reading();
+    ch4Reading();
+    //pdReading();
   }
 
-  if(timing%210==0)
-  {
-    
-  }
-  
   //Read from RoveComm
-  rovecomm_packet packet = RoveComm.read();
-  if(packet.data_id!=0)
+  packet = RoveComm.read();
+  if(packet.data_id != 0)
   {
-    switch (packet.data_id)
+    switch(packet.data_id)
     {
-      case RC_SCIENCESENSORSBOARD_UVLEDCONTROL_DATA_ID: //Switch UVLED on or off
+      case RC_SCIENCESENSORSBOARD_LIGHTS_DATA_ID:    //Switch UVLED on or off
         updateLed((int)packet.data[0]);
         break;
+      case RC_SCIENCESENSORSBOARD_FLASERS_DATA_ID:
+        uint8_t* lasersOn = (uint8_t*)packet.data;
+        if(lasersOn[0] != 0)
+        {
+          digitalWrite(Laser1, HIGH);
+          digitalWrite(Laser2, HIGH);
+          digitalWrite(Laser3, HIGH); 
+        }
```

Comment:
The FLasers command is a bitmasked value which controls the lasers individually. How it is now, the lasers can only be controlled all at once.